ivy_tests/test_ivy/test_frontends/test_torch/test_indexing_slicing_joining_mutating_ops.py

```diff
@@ -70,3 +71,61 @@ def test_torch_cat(
         dim=unique_idx,
         out=None,
     )
+
+    
+# composite strategy for input dimensions as permutations for frontends.torch.permute
+@st.composite
+def _permute_helper(draw):
+    shape = draw(st.shared(
+        helpers.get_shape(min_num_dims=1),
+        key='shape_value'
+    ))
+    dims = [x for x in range(len(shape))]
+    permutation = draw(st.permutations(dims))
+    return permutation
+
+
+# permute
+@given(
+    dtype_value=helpers.dtype_values_axis(
+        available_dtypes=tuple(
+            set(ivy_np.valid_float_dtypes).intersection(
+                set(ivy_torch.valid_float_dtypes))),
+        shape=helpers.get_shape(min_num_dims=1),
+            
+    ),
+    as_variable=st.booleans(),
+    num_positional_args=helpers.num_positional_args(
+        fn_name="ivy.functional.frontends.torch.permute"),
+    native_array=st.booleans(),
+    container=st.booleans(),
+    instance_method=st.booleans(),
+    with_out=st.booleans(),
+)
+def test_permute(
+        *,
+        dtype_values_axis,
+        permutation,
+        as_variable,
+        num_positional_args,
+        native_array,
+        container,
+        instance_method,
+        with_out,
+        fw,
+):
+    dtype, value, axis = dtype_values_axis
+    helpers.test_frontend_function(
```

Comment:
please read the description of helpers.test_frontend_function here:
"""Tests a frontend function for the current backend by comparing the result with
    the function in the associated framework.

    Parameters
    ----------
    input_dtypes
        data types of the input arguments in order.
    as_variable_flags
        dictates whether the corresponding input argument should be treated
        as an ivy Variable.
    with_out
        if True, the function is also tested with the optional out argument.
    num_positional_args
        number of input arguments that must be passed as positional
        arguments.
    native_array_flags
        dictates whether the corresponding input argument should be treated
        as a native array.
    fw
        current backend (framework).
    frontend
        current frontend (framework).
    fn_name
        name of the function to test.
    rtol
        relative tolerance value.
    atol
        absolute tolerance value.
    test_values
        if True, test for the correctness of the resulting values.
    all_as_kwargs_np
        input arguments to the function as keyword arguments.

    Returns
    -------
    ret
        optional, return value from the function
    ret_np
        optional, return value from the Numpy function
    """
You should not randomly generate the as_variable_flags, with_out, native_array_flags, container_flags and instance_method arguments as they should be determined according to the situation like this example here:
@given(
    dtype_and_x=helpers.dtype_and_values(available_dtypes=ivy_np.valid_float_dtypes),
    num_positional_args=helpers.num_positional_args(fn_name="variable"),
    data=st.data(),
)
@handle_cmd_line_args
def test_variable(
    *,
    dtype_and_x,
    num_positional_args,
    native_array,
    container,
    instance_method,
    fw,
):
    dtype, x = dtype_and_x
    helpers.test_function(
        input_dtypes=dtype,
        as_variable_flags=True,
        with_out=False,
        num_positional_args=num_positional_args,
        native_array_flags=native_array,
        container_flags=container,
        instance_method=instance_method,
        fw=fw,
        fn_name="variable",
        x=np.asarray(x, dtype=dtype),
    )

Reply:
I see, how would you set them? would these be correct?
`as_variable=helpers.array_bools()`
`with_out`: for function that has this optional out, what should I set it to? 
`native_array=helpers.array_bools(),` 
how should I set these in this case? `container_flags` and `instance_method`

Reply:
In this case, the permute function does have an out argument, so the with_out argument should always be True.
Finally, as_variable_flags, native_array_flags should be set automatically, by adding @handle_cmd_line_args before the test function and have arguments in your test function receiving the values and passing into test_frontend_function.